### PR TITLE
feat: add fp16 inference support (torch/onnx)

### PR DESCRIPTION
**Pull request by OrangeSodahub**
- [x] clip_torch
- [x] clip_onnx

### CODE REVIEW

File: server/clip_server/helper.py [hunk 1]
```diff
@@ -19,6 +20,9 @@
 )
 
 
+__cast_dtype__ = {'fp16': torch.float16, 'fp32': torch.float32, 'bf16': torch.bfloat16}
+
+
```
Comment (ZiniuYu) [inline server/clip_server/helper.py:25]: do not need this part

File: tests/test_fp16.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+            ]
+        ),
+        DocumentArray.from_files(
+            f'{os.path.dirname(os.path.abspath(__file__))}/**/*.jpg'
+        ),
+    ],
+)
+def test_docarray_inputs(make_flow, inputs):
+    c = Client(server=f'grpc://0.0.0.0:{make_flow.port}')
+    r = c.encode(inputs if not callable(inputs) else inputs())
```
Comment (jemmyshin) [inline tests/test_fp16.py:58]:
```suggestion
    r = c.encode(inputs if not callable(inputs) else inputs())
```

please check the dtype of embedding here

**Review by jemmyshin — CHANGES_REQUESTED** [review]

File: server/clip_server/helper.py (hunk 1 above)
Comment (OrangeSodahub) [reply server/clip_server/helper.py]: moved

File: tests/test_simple.py [hunk 3]
```diff
@@ -27,6 +27,7 @@ def test_protocols(port_generator, protocol, jit, pytestconfig):
         c.profile(content=f'{pytestconfig.rootdir}/tests/img/00000.jpg')
 
 
+@pytest.mark.gpu
```
[the remark this answers is not in the document]
Comment (OrangeSodahub) [reply tests/test_simple.py:30]: cpu and gpu are ok

**Review by jemmyshin — APPROVED** [review]
lgtm

**Pull request merged**